crates/cxx-qt-lib-headers/include/qt_types.h

```diff
@@ -24,6 +25,100 @@
 
 #include "rust/cxx.h"
 
+// Define which Qt types are relocatable
+template<>
+struct rust::IsRelocatable<QColor> : std::true_type
+{
+};
+static_assert(QTypeInfo<QColor>::isRelocatable);
+
+template<>
+struct rust::IsRelocatable<QDateTime> : std::true_type
+{
+};
+static_assert(QTypeInfo<QDateTime>::isRelocatable);
+
+template<>
+struct rust::IsRelocatable<QString> : std::true_type
+{
+};
+static_assert(QTypeInfo<QString>::isRelocatable);
+
+template<>
+struct rust::IsRelocatable<QUrl> : std::true_type
+{
+};
+static_assert(QTypeInfo<QUrl>::isRelocatable);
+
+template<>
+struct rust::IsRelocatable<QVariant> : std::true_type
+{
+};
+static_assert(QTypeInfo<QVariant>::isRelocatable);
+
+// Ensure that trivially copyable is correct
+// If this is false then we need to implement Clone rather than derive
+
+// QColor still had copy constructors in Qt 5 but they could have been trivial
+#if (QT_VERSION >= QT_VERSION_CHECK(6, 0, 0))
+static_assert(std::is_trivially_copyable<QColor>::value);
+#endif
+static_assert(!std::is_trivially_copyable<QDateTime>::value);
+static_assert(!std::is_trivially_copyable<QString>::value);
+static_assert(!std::is_trivially_copyable<QUrl>::value);
+static_assert(!std::is_trivially_copyable<QVariant>::value);
+
+// Ensure that trivially destructible is correct
+// If this is false then we need to implement Drop rather than derive
+static_assert(std::is_trivially_destructible<QColor>::value);
+static_assert(!std::is_trivially_destructible<QDateTime>::value);
+static_assert(!std::is_trivially_destructible<QString>::value);
+static_assert(!std::is_trivially_destructible<QUrl>::value);
+static_assert(!std::is_trivially_destructible<QVariant>::value);
+
+// Ensure that types have the alignment and size we are expecting
+static_assert(alignof(QColor) <= alignof(std::size_t[2]),
+              "unexpectedly large QColor alignment");
+static_assert(sizeof(QColor) == sizeof(std::size_t[2]),
+              "unexpected QColor size");
+
+// The layout has changed between Qt 5 and Qt 6
+//
+// Qt5 QString has one pointer as a member
+// Qt6 QString has one member, which contains two pointers and a size_t
+#if (QT_VERSION >= QT_VERSION_CHECK(6, 0, 0))
+static_assert(alignof(QString) <= alignof(std::size_t[3]),
+              "unexpectedly large QString alignment");
+static_assert(sizeof(QString) == sizeof(std::size_t[3]),
+              "unexpected QString size");
+#else
+static_assert(alignof(QString) <= alignof(std::size_t),
+              "unexpectedly large QString alignment");
+static_assert(sizeof(QString) == sizeof(std::size_t),
+              "unexpected QString size");
+#endif
+
+static_assert(alignof(QUrl) <= alignof(std::size_t),
```

Comment:
It would be helpful to have comments documenting where these values came from for each type like you've written for QString. Even better would be links to Qt implementations on https://codebrowser.dev/